compliance_checker/cf/cf_1_7.py

```diff
@@ -75,8 +75,47 @@ def __init__(self, options=None):
         self.grid_mapping_dict = grid_mapping_dict17
         self.grid_mapping_attr_types = grid_mapping_attr_types17
 
+    def check_external_variables(self, ds):
+        """
+        The global external_variables attribute is a blank-separated list of the
+        names of variables which are named by attributes in the file but which
+        are not present in the file. These variables are to be found in other
+        files (called "external files") but CF does not provide conventions for
+        identifying the files concerned. The only attribute for which CF
+        standardises the use of external variables is cell_measures.
+
+        :param netCDF4.Dataset ds: An open netCDF dataset
+        :rtype: compliance_checker.base.Result
+        """
+        external_vars_ctx = TestCtx(
+            BaseCheck.MEDIUM, self.section_titles["2.6.3"]
+        )
+        # IMPLEMENTATION CONFORMANCE 2.6.3 REQUIRED 2/2
+        try:
+            external_var_names = set(ds.external_variables.strip().split())
+
+            bad_external_var_names = (external_var_names.intersection(ds.variables))
+
+            if bad_external_var_names:
+                external_vars_ctx.out_of += 1
+                bad_msg = ("Global attribute external_variables should not "
+                           "have any variable names which are present in the dataset. "
+                           "Currently, the following names appear in both external_variables "
+                           f"and the dataset's variables: {bad_external_var_names}")
+                external_vars_ctx.messages.append(bad_msg)
+
+        # string/global attributes are handled in Appendix A checks
+        except (AttributeError, ValueError):
+            pass
```

Comment:
NB: currently one has to check the Appendix A attributes in CF checker to get an error on the type:

```
python cchecker.py -O cf:enable_appendix_a_checks -t cf /tmp/foo2.nc 
Running Compliance Checker on the datasets from: ['/tmp/foo2.nc']


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                Version unknown                                 
                     Report generated 2022-04-13T18:45:57Z                      
                                     cf:1.8                                     
http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html
--------------------------------------------------------------------------------
                               Corrective Actions                               
foo2.nc has 2 potential issues


                                     Errors                                     
--------------------------------------------------------------------------------
§2.6 Attributes
* external_variables must be a string

§3.3 Standard Name
* Attribute long_name or/and standard_name is highly recommended for variable ext3
```

Also note that the section does not correspond to 2.6.3.   Historical reasons for this were related to backwards compatibility, which I disagreed with, so an option was supplied for this behavior.  In any event, there is some redundancy in CF with respect to Appendix A specifying the types for global attributes as well as some checks calling for attribute types.  It'd be best to check this by default, but have it only checked once and have it grouped with the corresponding section.